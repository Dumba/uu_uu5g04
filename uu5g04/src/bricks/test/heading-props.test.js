/**
 * Copyright (C) 2019 Unicorn a.s.
 *
 * This program is free software; you can use it under the terms of the UAF Open License v01 or
 * any later version. The text of the license is available in the file LICENSE or at www.unicorn.com.
 *
 * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even
 * the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See LICENSE for more details.
 *
 * You may contact Unicorn a.s. at address: V Kapslovne 2767/2, Praha 3, Czech Republic or
 * at the email: info@unicorn.com.
 */

//@@viewOn:imports
import React from "react";
import UU5 from "uu5g04";
import "uu5g04-bricks";
import createReactClass from "create-react-class";
//@@viewOff:imports

const { mount, shallow, wait } = UU5.Test.Tools;

const MyHeadingHandler = createReactClass({
  getInitialState: () => {
    return {
      isCalled: false
    };
  },

  onScrollToFixedHandler(event) {
    alert("onScrollToFixed was called!");
    this.setState({ isCalled: true });
  },

  onScrollToBlockedHandler(event) {
    alert("onScrollToBlocked was called!");
    this.setState({ isCalled: true });
  },

  render() {
    return (
      <UU5.Bricks.Heading
        content="Bricks.Heading"
        style={{
          padding: "10px",
          backgroundColor: "cornflowerblue"
        }}
        fixedOnScroll={true}
        onScrollToFixed={this.onScrollToFixedHandler}
        onScrollToBlocked={this.onScrollToBlockedHandler}
      />
    );
  }
});

const CONFIG = {
  mixins: [
    "UU5.Common.BaseMixin",
    "UU5.Common.ElementaryMixin",
    "UU5.Common.ContentMixin",
    "UU5.Common.NestingLevelMixin",
    "UU5.Common.PureRenderMixin"
  ],
  props: {
    fixed: {
      values: [true, false]
    },
    fixedOnScroll: {
      values: [true, false]
    },
    hideOnScroll: {
      values: [true, false]
    },
    spaceHolder: {
      values: [true, false]
    }
    //onScrollToFixed:{},
    //onScrollToBlocked:{},
  },
  requiredProps: {},
  opt: {
    shallowOpt: {
      disableLifecycleMethods: false
    }
  }
};

describe(`UU5.Bricks.Heading`, () => {
  UU5.Test.Tools.testProperties(UU5.Bricks.Heading, CONFIG);
});

describe(`UU5.Bricks.Heading`, () => {
  it(`UU5.Bricks.Heading - onScrollToFixed()`, () => {
    window.alert = jest.fn();
    const wrapper = shallow(<MyHeadingHandler />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.state().isCalled).toBeFalsy();
    wrapper.simulate("scrollToFixed");
    expect(window.alert).toBeCalled();
    expect(window.alert).toHaveBeenCalledWith("onScrollToFixed was called!");
    expect(wrapper.state().isCalled).toBeTruthy();
    expect(window.alert.mock.calls[0][0]).toEqual("onScrollToFixed was called!");
    expect(wrapper).toMatchSnapshot();
  });

  it(`UU5.Bricks.Heading - onScrollToBlocked()`, () => {
    window.alert = jest.fn();
    const wrapper = shallow(<MyHeadingHandler />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.state().isCalled).toBeFalsy();
    wrapper.simulate("scrollToBlocked");
    expect(window.alert).toBeCalled();
    expect(window.alert).toHaveBeenCalledWith("onScrollToBlocked was called!");
    expect(wrapper.state().isCalled).toBeTruthy();
    expect(window.alert.mock.calls[0][0]).toEqual("onScrollToBlocked was called!");
    expect(wrapper).toMatchSnapshot();
  });
});

describe(`UU5.Bricks.Heading docKit examples`, () => {
  it(`UU5.Bricks.Heading example01`, () => {
    const wrapper = shallow(
      <UU5.Bricks.Container id={"uuID01"}>
        <UU5.Bricks.Heading
          id={"uuID"}
          content="Fixed"
          fixed
          style={{
            padding: "10px",
            backgroundColor: "cornflowerblue"
          }}
        />

        <UU5.Bricks.Paragraph id={"uuID02"} />
      </UU5.Bricks.Container>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
