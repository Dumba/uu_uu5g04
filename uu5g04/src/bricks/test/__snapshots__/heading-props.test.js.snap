// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UU5.Bricks.Heading UU5.Bricks.Heading - onScrollToBlocked() 1`] = `
<Heading
  className={null}
  content="Bricks.Heading"
  controlled={true}
  disabled={false}
  dynamic={false}
  fixed={false}
  fixedOnScroll={true}
  hidden={false}
  hideOnScroll={false}
  hideOnScrollOffset={64}
  id={null}
  ignoreInnerHTML={false}
  mainAttrs={null}
  name={null}
  nestingLevel={null}
  noIndex={false}
  onScrollToBlocked={[Function]}
  onScrollToFixed={[Function]}
  pureRender={false}
  ref_={null}
  selected={false}
  spaceHolder={false}
  style={
    Object {
      "backgroundColor": "cornflowerblue",
      "padding": "10px",
    }
  }
  tooltip={null}
/>
`;

exports[`UU5.Bricks.Heading UU5.Bricks.Heading - onScrollToBlocked() 2`] = `
<Heading
  className={null}
  content="Bricks.Heading"
  controlled={true}
  disabled={false}
  dynamic={false}
  fixed={false}
  fixedOnScroll={true}
  hidden={false}
  hideOnScroll={false}
  hideOnScrollOffset={64}
  id={null}
  ignoreInnerHTML={false}
  mainAttrs={null}
  name={null}
  nestingLevel={null}
  noIndex={false}
  onScrollToBlocked={[Function]}
  onScrollToFixed={[Function]}
  pureRender={false}
  ref_={null}
  selected={false}
  spaceHolder={false}
  style={
    Object {
      "backgroundColor": "cornflowerblue",
      "padding": "10px",
    }
  }
  tooltip={null}
/>
`;

exports[`UU5.Bricks.Heading UU5.Bricks.Heading - onScrollToFixed() 1`] = `
<Heading
  className={null}
  content="Bricks.Heading"
  controlled={true}
  disabled={false}
  dynamic={false}
  fixed={false}
  fixedOnScroll={true}
  hidden={false}
  hideOnScroll={false}
  hideOnScrollOffset={64}
  id={null}
  ignoreInnerHTML={false}
  mainAttrs={null}
  name={null}
  nestingLevel={null}
  noIndex={false}
  onScrollToBlocked={[Function]}
  onScrollToFixed={[Function]}
  pureRender={false}
  ref_={null}
  selected={false}
  spaceHolder={false}
  style={
    Object {
      "backgroundColor": "cornflowerblue",
      "padding": "10px",
    }
  }
  tooltip={null}
/>
`;

exports[`UU5.Bricks.Heading UU5.Bricks.Heading - onScrollToFixed() 2`] = `
<Heading
  className={null}
  content="Bricks.Heading"
  controlled={true}
  disabled={false}
  dynamic={false}
  fixed={false}
  fixedOnScroll={true}
  hidden={false}
  hideOnScroll={false}
  hideOnScrollOffset={64}
  id={null}
  ignoreInnerHTML={false}
  mainAttrs={null}
  name={null}
  nestingLevel={null}
  noIndex={false}
  onScrollToBlocked={[Function]}
  onScrollToFixed={[Function]}
  pureRender={false}
  ref_={null}
  selected={false}
  spaceHolder={false}
  style={
    Object {
      "backgroundColor": "cornflowerblue",
      "padding": "10px",
    }
  }
  tooltip={null}
/>
`;

exports[`UU5.Bricks.Heading checkGrammar 1`] = `
<Fragment>
  <div
    className="uu5-bricks-heading"
    id="uuID"
    key="main"
    style={Object {}}
    title={null}
  />
</Fragment>
`;

exports[`UU5.Bricks.Heading checkGrammar 2`] = `
<Fragment>
  <div
    className="uu5-bricks-heading"
    id="uuID"
    key="main"
    style={Object {}}
    title={null}
  />
</Fragment>
`;

exports[`UU5.Bricks.Heading checkGrammar 3`] = `
<Fragment>
  <div
    className="uu5-bricks-heading"
    id="uuID"
    key="main"
    style={Object {}}
    title={null}
  />
</Fragment>
`;

exports[`UU5.Bricks.Heading checkGrammar 4`] = `
<Fragment>
  <div
    className="uu5-bricks-heading"
    id="uuID"
    key="main"
    style={Object {}}
    title={null}
  />
</Fragment>
`;

exports[`UU5.Bricks.Heading checkHighlight 1`] = `
<Fragment>
  <div
    className="uu5-bricks-heading"
    id="uuID"
    key="main"
    style={Object {}}
    title={null}
  />
</Fragment>
`;

exports[`UU5.Bricks.Heading checkHighlight 2`] = `
<Fragment>
  <div
    className="uu5-bricks-heading"
    id="uuID"
    key="main"
    style={Object {}}
    title={null}
  />
</Fragment>
`;

exports[`UU5.Bricks.Heading checkHighlight 3`] = `
<Fragment>
  <div
    className="uu5-bricks-heading"
    id="uuID"
    key="main"
    style={Object {}}
    title={null}
  />
</Fragment>
`;

exports[`UU5.Bricks.Heading checkHighlight 4`] = `
<Fragment>
  <div
    className="uu5-bricks-heading"
    id="uuID"
    key="main"
    style={Object {}}
    title={null}
  />
</Fragment>
`;

exports[`UU5.Bricks.Heading checkSpaces 1`] = `
<Fragment>
  <div
    className="uu5-bricks-heading"
    id="uuID"
    key="main"
    style={Object {}}
    title={null}
  />
</Fragment>
`;

exports[`UU5.Bricks.Heading checkSpaces 2`] = `
<Fragment>
  <div
    className="uu5-bricks-heading"
    id="uuID"
    key="main"
    style={Object {}}
    title={null}
  />
</Fragment>
`;

exports[`UU5.Bricks.Heading checkSpaces 3`] = `
<Fragment>
  <div
    className="uu5-bricks-heading"
    id="uuID"
    key="main"
    style={Object {}}
    title={null}
  />
</Fragment>
`;

exports[`UU5.Bricks.Heading checkSpaces 4`] = `
<Fragment>
  <div
    className="uu5-bricks-heading"
    id="uuID"
    key="main"
    style={Object {}}
    title={null}
  />
</Fragment>
`;

exports[`UU5.Bricks.Heading className 1`] = `
<Fragment>
  <div
    className="uu5-bricks-heading my-classname"
    id="uuID"
    key="main"
    style={Object {}}
    title={null}
  />
</Fragment>
`;

exports[`UU5.Bricks.Heading content 1`] = `
<Fragment>
  <div
    className="uu5-bricks-heading"
    id="uuID"
    key="main"
    style={Object {}}
    title={null}
  >
    12345
  </div>
</Fragment>
`;

exports[`UU5.Bricks.Heading content 2`] = `
<Fragment>
  <div
    className="uu5-bricks-heading"
    id="uuID"
    key="main"
    style={Object {}}
    title={null}
  >
    <UU5.Common.TextCorrectorContextConsumer
      key="0"
      text="This is Content from ContentMixin"
    />
  </div>
</Fragment>
`;

exports[`UU5.Bricks.Heading content 3`] = `
<Fragment>
  <div
    className="uu5-bricks-heading"
    id="uuID"
    key="main"
    style={Object {}}
    title={null}
  >
    <UU5.Common.TextCorrectorContextConsumer
      key="0"
      text="This is Content from ContentMixin"
    />
  </div>
</Fragment>
`;

exports[`UU5.Bricks.Heading content 4`] = `
<Fragment>
  <div
    className="uu5-bricks-heading"
    id="uuID"
    key="main"
    style={Object {}}
    title={null}
  >
    12345
  </div>
</Fragment>
`;

exports[`UU5.Bricks.Heading controlled 1`] = `
<Fragment>
  <div
    className="uu5-bricks-heading"
    id="uuID"
    key="main"
    style={Object {}}
    title={null}
  />
</Fragment>
`;

exports[`UU5.Bricks.Heading controlled 2`] = `
<Fragment>
  <div
    className="uu5-bricks-heading"
    id="uuID"
    key="main"
    style={Object {}}
    title={null}
  />
</Fragment>
`;

exports[`UU5.Bricks.Heading controlled 3`] = `
<Fragment>
  <div
    className="uu5-bricks-heading"
    id="uuID"
    key="main"
    style={Object {}}
    title={null}
  />
</Fragment>
`;

exports[`UU5.Bricks.Heading controlled 4`] = `
<Fragment>
  <div
    className="uu5-bricks-heading"
    id="uuID"
    key="main"
    style={Object {}}
    title={null}
  />
</Fragment>
`;

exports[`UU5.Bricks.Heading default props 1`] = `
<Fragment>
  <div
    className="uu5-bricks-heading"
    id="uuID"
    key="main"
    style={Object {}}
    title={null}
  />
</Fragment>
`;

exports[`UU5.Bricks.Heading disabled 1`] = `
<Fragment>
  <div
    className="uu5-bricks-heading uu5-common-disabled"
    id="uuID"
    key="main"
    style={Object {}}
    title={null}
  >
    <span
      className="uu5-common-disabled-cover"
      key="disabledCover"
    />
  </div>
</Fragment>
`;

exports[`UU5.Bricks.Heading disabled 2`] = `
<Fragment>
  <div
    className="uu5-bricks-heading"
    id="uuID"
    key="main"
    style={Object {}}
    title={null}
  />
</Fragment>
`;

exports[`UU5.Bricks.Heading disabled 3`] = `
<Fragment>
  <div
    className="uu5-bricks-heading"
    id="uuID"
    key="main"
    style={Object {}}
    title={null}
  />
</Fragment>
`;

exports[`UU5.Bricks.Heading disabled 4`] = `
<Fragment>
  <div
    className="uu5-bricks-heading uu5-common-disabled"
    id="uuID"
    key="main"
    style={Object {}}
    title={null}
  >
    <span
      className="uu5-common-disabled-cover"
      key="disabledCover"
    />
  </div>
</Fragment>
`;

exports[`UU5.Bricks.Heading docKit examples UU5.Bricks.Heading example01 1`] = `
<div
  className="uu5-bricks-container uu5-bricks-container-spacing"
  id="uuID01"
  style={Object {}}
  title={null}
>
  <Heading
    className={null}
    content="Fixed"
    controlled={true}
    disabled={false}
    dynamic={false}
    fixed={true}
    fixedOnScroll={false}
    hidden={false}
    hideOnScroll={false}
    hideOnScrollOffset={64}
    id="uuID"
    ignoreInnerHTML={false}
    key="uuID"
    mainAttrs={null}
    name={null}
    nestingLevel={null}
    noIndex={false}
    onScrollToBlocked={null}
    onScrollToFixed={null}
    pureRender={false}
    ref_={null}
    selected={false}
    spaceHolder={false}
    style={
      Object {
        "backgroundColor": "cornflowerblue",
        "padding": "10px",
      }
    }
    tooltip={null}
  />
  <Paragraph
    className={null}
    content={null}
    controlled={true}
    disabled={false}
    dynamic={false}
    hidden={false}
    id="uuID02"
    ignoreInnerHTML={false}
    key="uuID02"
    mainAttrs={null}
    name={null}
    nestingLevel={null}
    noIndex={false}
    pureRender={false}
    ref_={null}
    selected={false}
    style={null}
    tooltip={null}
  />
</div>
`;

exports[`UU5.Bricks.Heading dynamic 1`] = `
<Fragment>
  <div
    className="uu5-bricks-heading"
    id="uuID"
    key="main"
    style={Object {}}
    title={null}
  />
</Fragment>
`;

exports[`UU5.Bricks.Heading dynamic 2`] = `
<Fragment>
  <div
    className="uu5-bricks-heading"
    id="uuID"
    key="main"
    style={Object {}}
    title={null}
  />
</Fragment>
`;

exports[`UU5.Bricks.Heading dynamic 3`] = `
<Fragment>
  <div
    className="uu5-bricks-heading"
    id="uuID"
    key="main"
    style={Object {}}
    title={null}
  />
</Fragment>
`;

exports[`UU5.Bricks.Heading dynamic 4`] = `
<Fragment>
  <div
    className="uu5-bricks-heading"
    id="uuID"
    key="main"
    style={Object {}}
    title={null}
  />
</Fragment>
`;

exports[`UU5.Bricks.Heading fixed 1`] = `
<Fragment>
  <div
    className="uu5-bricks-heading uu5-bricks-heading-fixed"
    id="uuID"
    key="main"
    style={Object {}}
    title={null}
  />
  <div
    className="uu5-bricks-heading-ghost"
    key="ghost"
    style={
      Object {
        "height": null,
      }
    }
  />
</Fragment>
`;

exports[`UU5.Bricks.Heading fixed 2`] = `
<Fragment>
  <div
    className="uu5-bricks-heading uu5-bricks-heading-fixed"
    id="uuID"
    key="main"
    style={Object {}}
    title={null}
  />
  <div
    className="uu5-bricks-heading-ghost"
    key="ghost"
    style={
      Object {
        "height": null,
      }
    }
  />
</Fragment>
`;

exports[`UU5.Bricks.Heading fixed 3`] = `
<Fragment>
  <div
    className="uu5-bricks-heading"
    id="uuID"
    key="main"
    style={Object {}}
    title={null}
  />
</Fragment>
`;

exports[`UU5.Bricks.Heading fixed 4`] = `
<Fragment>
  <div
    className="uu5-bricks-heading"
    id="uuID"
    key="main"
    style={Object {}}
    title={null}
  />
</Fragment>
`;

exports[`UU5.Bricks.Heading fixedOnScroll 1`] = `
<Fragment>
  <div
    className="uu5-bricks-heading"
    id="uuID"
    key="main"
    style={Object {}}
    title={null}
  />
</Fragment>
`;

exports[`UU5.Bricks.Heading fixedOnScroll 2`] = `
<Fragment>
  <div
    className="uu5-bricks-heading"
    id="uuID"
    key="main"
    style={Object {}}
    title={null}
  />
</Fragment>
`;

exports[`UU5.Bricks.Heading fixedOnScroll 3`] = `
<Fragment>
  <div
    className="uu5-bricks-heading"
    id="uuID"
    key="main"
    style={Object {}}
    title={null}
  />
</Fragment>
`;

exports[`UU5.Bricks.Heading fixedOnScroll 4`] = `
<Fragment>
  <div
    className="uu5-bricks-heading"
    id="uuID"
    key="main"
    style={Object {}}
    title={null}
  />
</Fragment>
`;

exports[`UU5.Bricks.Heading hidden 1`] = `
<Fragment>
  <div
    className="uu5-bricks-heading uu5-common-hidden"
    id="uuID"
    key="main"
    style={Object {}}
    title={null}
  />
</Fragment>
`;

exports[`UU5.Bricks.Heading hidden 2`] = `
<Fragment>
  <div
    className="uu5-bricks-heading"
    id="uuID"
    key="main"
    style={Object {}}
    title={null}
  />
</Fragment>
`;

exports[`UU5.Bricks.Heading hidden 3`] = `
<Fragment>
  <div
    className="uu5-bricks-heading"
    id="uuID"
    key="main"
    style={Object {}}
    title={null}
  />
</Fragment>
`;

exports[`UU5.Bricks.Heading hidden 4`] = `
<Fragment>
  <div
    className="uu5-bricks-heading uu5-common-hidden"
    id="uuID"
    key="main"
    style={Object {}}
    title={null}
  />
</Fragment>
`;

exports[`UU5.Bricks.Heading hideOnScroll 1`] = `
<Fragment>
  <div
    className="uu5-bricks-heading"
    id="uuID"
    key="main"
    style={Object {}}
    title={null}
  />
</Fragment>
`;

exports[`UU5.Bricks.Heading hideOnScroll 2`] = `
<Fragment>
  <div
    className="uu5-bricks-heading"
    id="uuID"
    key="main"
    style={Object {}}
    title={null}
  />
</Fragment>
`;

exports[`UU5.Bricks.Heading hideOnScroll 3`] = `
<Fragment>
  <div
    className="uu5-bricks-heading"
    id="uuID"
    key="main"
    style={Object {}}
    title={null}
  />
</Fragment>
`;

exports[`UU5.Bricks.Heading hideOnScroll 4`] = `
<Fragment>
  <div
    className="uu5-bricks-heading"
    id="uuID"
    key="main"
    style={Object {}}
    title={null}
  />
</Fragment>
`;

exports[`UU5.Bricks.Heading id 1`] = `
<Fragment>
  <div
    className="uu5-bricks-heading"
    id="myId"
    key="main"
    style={Object {}}
    title={null}
  />
</Fragment>
`;

exports[`UU5.Bricks.Heading ignoreHTML 1`] = `
<Fragment>
  <div
    className="uu5-bricks-heading"
    id="uuID"
    key="main"
    style={Object {}}
    title={null}
  />
</Fragment>
`;

exports[`UU5.Bricks.Heading ignoreHTML 2`] = `
<Fragment>
  <div
    className="uu5-bricks-heading"
    id="uuID"
    key="main"
    style={Object {}}
    title={null}
  />
</Fragment>
`;

exports[`UU5.Bricks.Heading ignoreHTML 3`] = `
<Fragment>
  <div
    className="uu5-bricks-heading"
    id="uuID"
    key="main"
    style={Object {}}
    title={null}
  />
</Fragment>
`;

exports[`UU5.Bricks.Heading ignoreHTML 4`] = `
<Fragment>
  <div
    className="uu5-bricks-heading"
    id="uuID"
    key="main"
    style={Object {}}
    title={null}
  />
</Fragment>
`;

exports[`UU5.Bricks.Heading mainAttrs 1`] = `
<Fragment>
  <div
    className="uu5-bricks-heading"
    id="uuID"
    key="main"
    style={Object {}}
    title={null}
  />
</Fragment>
`;

exports[`UU5.Bricks.Heading name 1`] = `
<Fragment>
  <div
    className="uu5-bricks-heading"
    id="uuID"
    key="main"
    name="myName"
    style={Object {}}
    title={null}
  />
</Fragment>
`;

exports[`UU5.Bricks.Heading nestingLevel 1`] = `
<Fragment>
  <div
    className="uu5-bricks-heading"
    id="uuID"
    key="main"
    style={Object {}}
    title={null}
  />
</Fragment>
`;

exports[`UU5.Bricks.Heading nestingLevel 2`] = `
<Fragment>
  <div
    className="uu5-bricks-heading"
    id="uuID"
    key="main"
    style={Object {}}
    title={null}
  />
</Fragment>
`;

exports[`UU5.Bricks.Heading nestingLevel 3`] = `
<Fragment>
  <div
    className="uu5-bricks-heading"
    id="uuID"
    key="main"
    style={Object {}}
    title={null}
  />
</Fragment>
`;

exports[`UU5.Bricks.Heading nestingLevel 4`] = `
<Fragment>
  <div
    className="uu5-bricks-heading"
    id="uuID"
    key="main"
    style={Object {}}
    title={null}
  />
</Fragment>
`;

exports[`UU5.Bricks.Heading nestingLevel 5`] = `
<Fragment>
  <div
    className="uu5-bricks-heading"
    id="uuID"
    key="main"
    style={Object {}}
    title={null}
  />
</Fragment>
`;

exports[`UU5.Bricks.Heading nestingLevel 6`] = `
<Fragment>
  <div
    className="uu5-bricks-heading"
    id="uuID"
    key="main"
    style={Object {}}
    title={null}
  />
</Fragment>
`;

exports[`UU5.Bricks.Heading nestingLevel 7`] = `
<Fragment>
  <div
    className="uu5-bricks-heading"
    id="uuID"
    key="main"
    style={Object {}}
    title={null}
  />
</Fragment>
`;

exports[`UU5.Bricks.Heading nestingLevel 8`] = `
<Fragment>
  <div
    className="uu5-bricks-heading"
    id="uuID"
    key="main"
    style={Object {}}
    title={null}
  />
</Fragment>
`;

exports[`UU5.Bricks.Heading nestingLevel 9`] = `
<Fragment>
  <div
    className="uu5-bricks-heading"
    id="uuID"
    key="main"
    style={Object {}}
    title={null}
  />
</Fragment>
`;

exports[`UU5.Bricks.Heading nestingLevel 10`] = `
<Fragment>
  <div
    className="uu5-bricks-heading"
    id="uuID"
    key="main"
    style={Object {}}
    title={null}
  />
</Fragment>
`;

exports[`UU5.Bricks.Heading nestingLevel 11`] = `
<Fragment>
  <div
    className="uu5-bricks-heading"
    id="uuID"
    key="main"
    style={Object {}}
    title={null}
  />
</Fragment>
`;

exports[`UU5.Bricks.Heading nestingLevel 12`] = `
<Fragment>
  <div
    className="uu5-bricks-heading"
    id="uuID"
    key="main"
    style={Object {}}
    title={null}
  />
</Fragment>
`;

exports[`UU5.Bricks.Heading nestingLevel 13`] = `
<Fragment>
  <div
    className="uu5-bricks-heading"
    id="uuID"
    key="main"
    style={Object {}}
    title={null}
  />
</Fragment>
`;

exports[`UU5.Bricks.Heading nestingLevel 14`] = `
<Fragment>
  <div
    className="uu5-bricks-heading"
    id="uuID"
    key="main"
    style={Object {}}
    title={null}
  />
</Fragment>
`;

exports[`UU5.Bricks.Heading nestingLevel 15`] = `""`;

exports[`UU5.Bricks.Heading nestingLevel 16`] = `""`;

exports[`UU5.Bricks.Heading nestingLevel 17`] = `""`;

exports[`UU5.Bricks.Heading nestingLevel 18`] = `""`;

exports[`UU5.Bricks.Heading nestingLevel 19`] = `""`;

exports[`UU5.Bricks.Heading nestingLevel 20`] = `""`;

exports[`UU5.Bricks.Heading noIndex 1`] = `
<Fragment>
  <div
    className="uu5-bricks-heading uu5-noindex"
    id="uuID"
    key="main"
    style={Object {}}
    title={null}
  />
</Fragment>
`;

exports[`UU5.Bricks.Heading noIndex 2`] = `
<Fragment>
  <div
    className="uu5-bricks-heading"
    id="uuID"
    key="main"
    style={Object {}}
    title={null}
  />
</Fragment>
`;

exports[`UU5.Bricks.Heading noIndex 3`] = `
<Fragment>
  <div
    className="uu5-bricks-heading"
    id="uuID"
    key="main"
    style={Object {}}
    title={null}
  />
</Fragment>
`;

exports[`UU5.Bricks.Heading noIndex 4`] = `
<Fragment>
  <div
    className="uu5-bricks-heading uu5-noindex"
    id="uuID"
    key="main"
    style={Object {}}
    title={null}
  />
</Fragment>
`;

exports[`UU5.Bricks.Heading pureRender 1`] = `
<Fragment>
  <div
    className="uu5-bricks-heading"
    id="uuID"
    key="main"
    style={Object {}}
    title={null}
  />
</Fragment>
`;

exports[`UU5.Bricks.Heading pureRender 2`] = `
<Fragment>
  <div
    className="uu5-bricks-heading"
    id="uuID"
    key="main"
    style={Object {}}
    title={null}
  />
</Fragment>
`;

exports[`UU5.Bricks.Heading pureRender 3`] = `
<Fragment>
  <div
    className="uu5-bricks-heading"
    id="uuID"
    key="main"
    style={Object {}}
    title={null}
  />
</Fragment>
`;

exports[`UU5.Bricks.Heading pureRender 4`] = `
<Fragment>
  <div
    className="uu5-bricks-heading"
    id="uuID"
    key="main"
    style={Object {}}
    title={null}
  />
</Fragment>
`;

exports[`UU5.Bricks.Heading selected 1`] = `
<Fragment>
  <div
    className="uu5-bricks-heading uu5-common-selected"
    id="uuID"
    key="main"
    style={Object {}}
    title={null}
  />
</Fragment>
`;

exports[`UU5.Bricks.Heading selected 2`] = `
<Fragment>
  <div
    className="uu5-bricks-heading"
    id="uuID"
    key="main"
    style={Object {}}
    title={null}
  />
</Fragment>
`;

exports[`UU5.Bricks.Heading selected 3`] = `
<Fragment>
  <div
    className="uu5-bricks-heading"
    id="uuID"
    key="main"
    style={Object {}}
    title={null}
  />
</Fragment>
`;

exports[`UU5.Bricks.Heading selected 4`] = `
<Fragment>
  <div
    className="uu5-bricks-heading uu5-common-selected"
    id="uuID"
    key="main"
    style={Object {}}
    title={null}
  />
</Fragment>
`;

exports[`UU5.Bricks.Heading spaceHolder 1`] = `
<Fragment>
  <div
    className="uu5-bricks-heading"
    id="uuID"
    key="main"
    style={Object {}}
    title={null}
  />
</Fragment>
`;

exports[`UU5.Bricks.Heading spaceHolder 2`] = `
<Fragment>
  <div
    className="uu5-bricks-heading"
    id="uuID"
    key="main"
    style={Object {}}
    title={null}
  />
</Fragment>
`;

exports[`UU5.Bricks.Heading spaceHolder 3`] = `
<Fragment>
  <div
    className="uu5-bricks-heading"
    id="uuID"
    key="main"
    style={Object {}}
    title={null}
  />
</Fragment>
`;

exports[`UU5.Bricks.Heading spaceHolder 4`] = `
<Fragment>
  <div
    className="uu5-bricks-heading"
    id="uuID"
    key="main"
    style={Object {}}
    title={null}
  />
</Fragment>
`;

exports[`UU5.Bricks.Heading style 1`] = `
<Fragment>
  <div
    className="uu5-bricks-heading"
    id="uuID"
    key="main"
    style={
      Object {
        "borderRadius": "2px",
        "padding": "8px",
      }
    }
    title={null}
  />
</Fragment>
`;

exports[`UU5.Bricks.Heading style 2`] = `
<Fragment>
  <div
    className="uu5-bricks-heading"
    id="uuID"
    key="main"
    style={
      Object {
        "backgroundColor": "red",
        "color": "blue",
      }
    }
    title={null}
  />
</Fragment>
`;

exports[`UU5.Bricks.Heading style 3`] = `
<Fragment>
  <div
    className="uu5-bricks-heading"
    id="uuID"
    key="main"
    style={
      Object {
        "backgroundColor": "red",
        "color": "blue",
      }
    }
    title={null}
  />
</Fragment>
`;

exports[`UU5.Bricks.Heading style 4`] = `
<Fragment>
  <div
    className="uu5-bricks-heading"
    id="uuID"
    key="main"
    style={
      Object {
        "borderRadius": "2px",
        "padding": "8px",
      }
    }
    title={null}
  />
</Fragment>
`;

exports[`UU5.Bricks.Heading textCorrector 1`] = `
<Fragment>
  <div
    className="uu5-bricks-heading"
    id="uuID"
    key="main"
    style={Object {}}
    title={null}
  />
</Fragment>
`;

exports[`UU5.Bricks.Heading textCorrector 2`] = `
<Fragment>
  <div
    className="uu5-bricks-heading"
    id="uuID"
    key="main"
    style={Object {}}
    title={null}
  />
</Fragment>
`;

exports[`UU5.Bricks.Heading textCorrector 3`] = `
<Fragment>
  <div
    className="uu5-bricks-heading"
    id="uuID"
    key="main"
    style={Object {}}
    title={null}
  />
</Fragment>
`;

exports[`UU5.Bricks.Heading textCorrector 4`] = `
<Fragment>
  <div
    className="uu5-bricks-heading"
    id="uuID"
    key="main"
    style={Object {}}
    title={null}
  />
</Fragment>
`;

exports[`UU5.Bricks.Heading tooltip 1`] = `
<Fragment>
  <div
    className="uu5-bricks-heading"
    id="uuID"
    key="main"
    style={Object {}}
    title="My Tooltip"
  />
</Fragment>
`;
